File: Cs/SampleApp/Form/ViewModel/InputDialog.cs
using AYam.Common.ViewModel;
using System;

namespace SampleApp.Form.ViewModel
{

    /// <summary>
    /// スクリーンキーボードサンプルApp
    /// </summary>
    public class InputDialog : VMBase, IDisposable
    {

        #region Property

        /// <summary>
        /// テンキー1プロパティ
        /// </summary>
        public string NumText1
        {
            get { return _Model.NumText1; }
            set
            {
                if (!_Model.NumText1.Equals(value))
                {
                    _Model.NumText1 = value;
                    CallPropertyChanged();
                }
            }
        }

        /// <summary>
        /// テンキー2プロパティ
        /// </summary>
        public string NumText2
        {
            get { return _Model.NumText2; }
            set
            {
                if (!_Model.NumText2.Equals(value))
                {
                    _Model.NumText2 = value;
                    CallPropertyChanged();
                }
            }
        }

        /// <summary>
        /// 文字入力1プロパティ
        /// </summary>
        public string StringText1
        {
            get { return _Model.StringText1; }
            set
            {
                if (!_Model.StringText1.Equals(value))
                {
                    _Model.StringText1 = value;
                    CallPropertyChanged();
                }
            }
        }

        /// <summary>
        /// 文字入力2プロパティ
        /// </summary>
        public string StringText2
        {
            get { return _Model.StringText2; }
            set
            {
                if (!_Model.StringText2.Equals(value))
                {
                    _Model.StringText2 = value;
                    CallPropertyChanged();
                }
            }
        }

        #endregion

        /// <summary>
        /// InputDialog.Model
        /// </summary>
        private Model.InputDialog _Model;

        /// <summary>
        /// InputDialog.ViewModel
        /// </summary>
        public InputDialog()
        {
            _Model = new Model.InputDialog();
        }

        /// <summary>
        /// 終了処理
        /// </summary>
        public void Dispose()
        {
            _Model.Dispose();
        }

    }

}

File: VB/ScreenKeyboard/SampleProject/Form/ViewModel/InputDialog.vb
Imports AYam.Common.ViewModel
Imports System

Namespace Form.ViewModel

    ''' <summary>
    ''' InputDialog.ViewModel
    ''' </summary>
    Public Class InputDialog
        Inherits VMBase
        Implements IDisposable

#Region "Property"

        ''' <summary>
        ''' テンキー１プロパティ
        ''' </summary>
        Public Property NumText1 As String
            Get
                Return _Model.NumText1
            End Get
            Set(value As String)

                If Not _Model.NumText1.Equals(value) Then

                    _Model.NumText1 = value
                    CallPropertyChanged()

                End If

            End Set
        End Property

        ''' <summary>
        ''' テンキー２プロパティ
        ''' </summary>
        Public Property NumText2 As String
            Get
                Return _Model.NumText2
            End Get
            Set(value As String)

                If Not _Model.NumText2.Equals(value) Then

                    _Model.NumText2 = value
                    CallPropertyChanged()

                End If

            End Set
        End Property

        ''' <summary>
        ''' 文字入力１プロパティ
        ''' </summary>
        Public Property StringText1 As String
            Get
                Return _Model.StringText1
            End Get
            Set(value As String)

                If Not _Model.StringText1.Equals(value) Then

                    _Model.StringText1 = value
                    CallPropertyChanged()

                End If

            End Set
        End Property

        ''' <summary>
        ''' 文字入力２プロパティ
        ''' </summary>
        Public Property StringText2 As String
            Get
                Return _Model.StringText2
            End Get
            Set(value As String)

                If Not _Model.StringText2.Equals(value) Then

                    _Model.StringText2 = value
                    CallPropertyChanged()

                End If

            End Set
        End Property

#End Region

        ''' <summary>
        ''' Model
        ''' </summary>
        Private _Model As Model.InputDialog

        ''' <summary>
        ''' InputDialog.ViewModel
        ''' </summary>
        Public Sub New()

            _Model = New Model.InputDialog()

        End Sub

        ''' <summary>
        ''' 終了処理
        ''' </summary>
        Public Sub Dispose() Implements IDisposable.Dispose
            _Model.Dispose()
            _Model = Nothing
        End Sub

    End Class

End Namespace

